src/Marten/Internal/Sessions/EventTracingConnectionLifetime.cs

```diff
@@ -0,0 +1,286 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Marten.Services;
+using Npgsql;
+
+namespace Marten.Internal.Sessions;
+
+internal class EventTracingConnectionLifetime :
+    IConnectionLifetime
+{
+    private const string MartenCommandExecutionStarted = "MartenCommandExecutionStarted";
+    private const string MartenCommandExecutionFailed = "MartenCommandExecutionFailed";
+    private const string MartenBatchExecutionStarted = "MartenBatchExecutionStarted";
+    private const string MartenBatchExecutionFailed = "MartenBatchExecutionFailed";
+    private const string MartenBatchPagesExecutionStarted = "MartenBatchPagesExecutionStarted";
+    private const string MartenBatchPagesExecutionFailed = "MartenBatchPagesExecutionFailed";
```

Comment:
According [OTel semantics](https://opentelemetry.io/docs/specs/semconv/general/events/) these should have different names which we should follow.
```suggestion
    private const string MartenCommandExecutionStarted = "marten.command.execution.started";
    private const string MartenCommandExecutionFailed = "marten.command.execution.failed";
    private const string MartenBatchExecutionStarted = "marten.batch.execution.started";
    private const string MartenBatchExecutionFailed = "marten.batch.execution.failed";
    private const string MartenBatchPagesExecutionStarted = "marten.batch.pages.execution.started";
    private const string MartenBatchPagesExecutionFailed = "marten.batch.pages.execution.failed";
```